src/main/webapp/app/assessment/assessment-instructions/expandable-section/expandable-section.component.ts

```diff
@@ -1,17 +1,34 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnInit } from '@angular/core';
 import { faAngleDown, faAngleRight } from '@fortawesome/free-solid-svg-icons';
+import { LocalStorageService } from 'ngx-webstorage';
 
 @Component({
     selector: 'jhi-expandable-section',
     templateUrl: './expandable-section.component.html',
 })
-export class ExpandableSectionComponent {
+export class ExpandableSectionComponent implements OnInit {
     @Input() headerKey: string;
-    @Input() isCollapsed = false;
     @Input() hasTranslation = true;
     @Input() isSubHeader = false;
 
+    isCollapsed: boolean;
     // Icons
     faAngleRight = faAngleRight;
     faAngleDown = faAngleDown;
+
+    constructor(private localStorageService: LocalStorageService) {}
+
+    ngOnInit(): void {
+        const collapsed = this.localStorageService.retrieve(this.headerKey);
+        this.isCollapsed = collapsed !== undefined && collapsed !== null && (collapsed as boolean);
+        this.localStorageService.store(this.headerKey, this.isCollapsed);
```

Comment:
Can we maybe add some kind of prefix here? To indicate that this value saves the collapsed-value?

Reply:
Good idea :+1: